app/src/main/java/com/m/androidNativeApp/MainActivity.java

```diff
@@ -37,26 +47,55 @@
 import static com.m.helper.LoginActivity.mobileService;
 
 
-public class MainActivity extends AppCompatActivity {
+
+public class MainActivity extends AppCompatActivity implements MessageHandler {
+
     public static ApolloClient client;
     private String taskTitle, taskDescription, taskId;
     private RecyclerView recyclerView;
     private ItemAdapter itemAdapter;
     private List<Item> itemList;
 
+    private String TAG = "Main Activity: APP:";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
         setupClient();
+        setupPush();
 
         itemAdapter = getAdapter();
-
         MainActivity.this.runOnUiThread(() -> recyclerView.setAdapter(itemAdapter));
 
         getTasks();
         subscribeToAddTask();
         subscribeToDeleteTask();
+        toastStartUpPushNotification();
+
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        RegistrarManager.registerMainThreadHandler(this); // 1
+        RegistrarManager.unregisterBackgroundThreadHandler(NotifyingHandler.instance);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        RegistrarManager.unregisterMainThreadHandler(this); // 2
```

Comment:
what are the comments (`1`, `2`) for?

Reply:
No idea. I might have put them in by mistake at some point and missed them when reading back over. I will remove them.